target/linux/pistachio/base-files/etc/init.d/csr

```diff
@@ -0,0 +1,206 @@
+#!/bin/sh /etc/rc.common
+#
+# Copyright 2016 by Imagination Technologies Limited and/or its affiliated group companies.
+#
+# All rights reserved.  No part of this software, either
+# material or conceptual may be copied or distributed,
+# transmitted, transcribed, stored in a retrieval system
+# or translated into any human or computer language in any
+# form by any means, electronic, mechanical, manual or
+# other-wise, or disclosed to the third parties without the
+# express written permission of Imagination Technologies
+# Limited, Home Park Estate, Kings Langley, Hertfordshire,
+# WD4 8LZ, U.K.
+
+START=80
+STOP=90
+
+BT_RST_N_GPIO=0
+{
+grep "Marduk" /proc/device-tree/model
+if [ $? -eq 0 ]; then
+	BT_RST_N_GPIO=43
+fi
+} > /dev/null 2>&1
+
+if [ $BT_RST_N_GPIO -eq 0 ]; then
+	echo -e "Unkown Platform"
+	exit 1
+fi
+
+setup_csr_gpio() {
+	# Set up BT_RST_N MFIO access
+	# check if the gpio has already been exported
+	{
+	if [ ! -d "/sys/class/gpio/gpio$BT_RST_N_GPIO" ];then
+		echo $BT_RST_N_GPIO > /sys/class/gpio/export
+	fi
+	} > /dev/null 2>&1
+	echo out > /sys/class/gpio/gpio$BT_RST_N_GPIO/direction
+}
+
+hide_csr_gpio() {
+        # Remove BT_RST_N MFIO access
+        # check if the gpio has already been exported
+        {
+        if [ -d "/sys/class/gpio/gpio$BT_RST_N_GPIO" ];then
+                echo $BT_RST_N_GPIO > /sys/class/gpio/unexport
+        fi
+        } > /dev/null 2>&1
+}
+
+
+n_reset_csr() {
+	setup_csr_gpio
+	echo 1 > /sys/class/gpio/gpio$BT_RST_N_GPIO/value
+	hide_csr_gpio
+}
+
+reset_csr() {
+	setup_csr_gpio
+	echo 0 > /sys/class/gpio/gpio$BT_RST_N_GPIO/value
+	hide_csr_gpio
+}
+
+
+bringup_csr() {
+local WIFI_STA_MAC=0x$(hexdump -n 6 -v -e '/1 "%02X"' /proc/device-tree/uccp@18480000/mac-address0)
+local BT_OTP_OFFSET=0x2
+local MAC=$(printf "%x" $((WIFI_STA_MAC + BT_OTP_OFFSET)))
+MAC=$(echo $BT_MAC | sed 's/.\{2\}/&:/g' | sed 's/.$//')
```

Comment:
Instead of sed, use macaddr_canonicalize function provided by /lib/functions/system.sh.